### ISSUE TITLE
How to resolve the optbinning issue

### ISSUE THREAD

**Issue by Jagadeesha89** [guillermo-navas-palencia/optbinning#328]
<img width="735" alt="Capture1" src="https://github.com/user-attachments/assets/b637f8cd-e8ce-4f3b-a0d1-ccf0648747bb">
              how to resolve this, because previously i ran the code to develop the scorecard its work well. this time i tried with same data but it gets stopped in the fit the binning process for X,y. Kindly help me in this

_Originally posted by @Jagadeesha89 in https://github.com/guillermo-navas-palencia/optbinning/issues/326#issuecomment-2244258260_

**Comment by guillermo-navas-palencia** [comment]
I would need much more information to say anything about this problem. You must provide a reproducible example.

**Comment by Jagadeesha89** [comment]
This is my code entire code blocks, when i try to fit pass this code "binning_process.fit(X,y)" every time my kernel gets stopped and popup for restart. I tried with python 3.11 version and 3.9 version.
<img width="843" alt="PART3" src="https://github.com/user-attachments/assets/ae29a892-303a-4603-89d2-ddad043d08f0">
<img width="838" alt="PARt2" src="https://github.com/user-attachments/assets/9f5308cb-43dc-4b0b-801d-d7ae54fb6dc2">
<img width="879" alt="Part1" src="https://github.com/user-attachments/assets/69dd2a89-fe72-460b-81b4-43acca9f5fee">

**Comment by pangjac** [comment]
How about `pip install -U numpy` ? Any recent dependent package versions you have upgraded recently?

**Comment by Jagadeesha89** [comment]
I am using 1.26.4 numpy package, i tried with python package with 3.9 version and latest version. I am getting same error

**Comment by Jagadeesha89** [comment]
if the number of rows in my data frame with in 20 it is working, if the data frame consist more than 20 rows jupyter note book kernel gets died. Please check is there any condition passed accessing the data.

**Comment by Jagadeesha89** [comment]
Even if i created the manual data with more than 20 rows with 2 columns its not working its not problem with data set. I have installed latest version of optbinning. Check with your end

**Comment by Jagadeesha89** [comment]
Any update

**Comment by bmreiniger** [comment]
Please add that example as code, so others can easily run it themselves. And any package versions or system specifications may also be helpful.

I agree (from the last issue) that this is unlikely to be an OptBinning problem, and if it's system-specific I don't know how much I could help. But at the very least, I'll run your example locally to verify whether it's your system.

**Comment by Jagadeesha89** [comment]
**Please find the below code which i used to run in my system**

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
pd.set_option('display.max_rows',150)
from optbinning import OptimalBinning  
from optbinning import Scorecard
from optbinning import BinningProcess
from optbinning.scorecard import plot_auc_roc, plot_cap, plot_ks
from sklearn.linear_model import LogisticRegression
from sklearn.pipeline import Pipeline
from sklearn.base import BaseEstimator, TransformerMixin

data=pd.read_csv(r"D:\CB_intigration\ScoreCard MFI Business\Smaple_data.csv")
[Smaple_data - Copy.csv](https://github.com/user-attachments/files/16450754/Smaple_data.-.Copy.csv)


x=data.drop('Status',axis=1)
y=data['Status']

variable_names = x.columns[:].tolist()
print(variable_names)

selection_criteria = {
    "iv": {"min": 0.07, "max": 0.5},
    "quality_score": {"min": 0.01}
}

binning_process = BinningProcess(variable_names, special_codes=[],
                                 selection_criteria=selection_criteria)

estimator = LogisticRegression(fit_intercept=True,solver='lbfgs')

binning_process.fit(x,y)

here i am getting the popup saying kernel appears to have died it will restart automatically.

sample data attached for your refrence.

**Comment by bmreiniger** [comment]
That runs fine in a Google Colab notebook.

**Comment by Jagadeesha89** [comment]
its working in python 3.11.5 version but >3.11.5 version python its throwing an error kernel died. I tried in VS code using virtual environment 3.11.5 python version its working but latest version of python is not supporting
